crates/ethrpc/src/http.rs

```diff
@@ -72,19 +72,33 @@ async fn execute_rpc<T: DeserializeOwned>(
 ) -> Result<T, Web3Error> {
     let body = serde_json::to_string(&request)?;
     tracing::trace!(name = %inner.name, %id, %body, "executing request");
-    let mut request = client
+    let mut request_builder = client
         .post(inner.url.clone())
         .header(header::CONTENT_TYPE, "application/json")
         .header("X-RPC-REQUEST-ID", id.to_string())
         .body(body);
-    if let Some(request_id) = observe::request_id::get_task_local_storage() {
-        request = request.header("X-REQUEST-ID", request_id);
+    match request {
+        Request::Single(Call::MethodCall(method)) => {
+            if let Some(request_id) = observe::request_id::get_task_local_storage() {
+                request_builder = request_builder.header("X-REQUEST-ID", request_id);
+            }
+            request_builder = request_builder.header("X-RPC-METHOD", method.method.clone());
+        }
+        Request::Batch(_) => {
+            if let Some(metadata) = observe::request_id::get_task_local_storage() {
+                request_builder = request_builder.header("X-RPC-METADATA", metadata);
```

Comment:
Why do we have to have different headers for single and batch requests?
Sounds like the data that is expressive enough for the batch case should also be good enough for the single case.

Reply:
IMO, for a single call and the absence of tracing_id, it makes sense to log the method name at least. Logging batched method names without trace ids doesn't seem to provide useful info.

Reply:
Let's not have a match on `request` here to use the same code path (and header) for single and batch requests. We probably only very rarely send individual requests so sending some (possibly) unnecessary metadata seems fine for the sake of code simplification.